x-pack/elastic-agent/pkg/core/plugin/app/app.go

```diff
@@ -121,66 +131,61 @@ func (a *Application) Stop() {
 	a.appLock.Lock()
 	defer a.appLock.Unlock()
 
-	if a.state.Status == state.Running && a.state.ProcessInfo != nil {
-		if closeClient, ok := a.grpcClient.(closer); ok {
-			closeClient.Close()
-		}
-
-		process.Stop(a.logger, a.state.ProcessInfo.PID)
+	if a.state.Status == state.Stopped {
+		return
+	}
 
-		a.state.Status = state.Stopped
-		a.state.ProcessInfo = nil
-		a.grpcClient = nil
-
-		// remove generated configuration if present
-		filename := fmt.Sprintf(configFileTempl, a.id)
-		filePath, err := filepath.Abs(filepath.Join(a.downloadConfig.InstallPath, filename))
-		if err == nil {
-			// ignoring error: not critical
-			os.Remove(filePath)
+	stopSig := os.Interrupt
+	if a.srvState != nil {
+		if err := a.srvState.Stop(a.processConfig.StopTimeout); err != nil {
+			// kill the process if stop through GRPC doesn't work
+			stopSig = os.Kill
+		}
+		a.srvState = nil
+	}
+	if a.state.ProcessInfo != nil {
+		if err := a.state.ProcessInfo.Process.Signal(stopSig); err == nil {
```

Comment:
If it doesn't stop "on time", I assume the Timeout from line 140 kicks in?

Reply:
Correct, and the process will be killed not terminated.